packages/commonwealth/shared/schemas/createCommunitySchema.ts

```diff
@@ -0,0 +1,83 @@
+import { EnsProvider } from '@ethersproject/providers/lib/base-provider';
+import {
+  ChainBase,
+  ChainCategoryType,
+  ChainNetwork,
+  ChainType,
+} from 'common-common/src/types';
+import { z } from 'zod';
+import { ALL_COMMUNITIES } from '../../server/middleware/databaseValidationService';
+import { getFileSizeBytes } from '../../server/util/getFilesSizeBytes';
+
+export const MAX_COMMUNITY_IMAGE_SIZE_KB = 500;
+
+async function checkIconSize(val, ctx) {
+  const fileSizeBytes = await getFileSizeBytes(val);
+  if (fileSizeBytes === 0) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      message: "Image url provided doesn't exist",
+    });
+  }
+  if (fileSizeBytes >= MAX_COMMUNITY_IMAGE_SIZE_KB * 1024) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      message: `Image must be smaller than ${MAX_COMMUNITY_IMAGE_SIZE_KB}kb`,
+    });
+  }
+}
+
+export const createCommunitySchema = (provider?: EnsProvider) =>
```

Comment:
@masvelio provider is optional, I am not sure how long it will take to import Web3 on client side. If it takes too long just skip it by passing in null and we will only validate the ENS on the server side.

Reply:
Disregard this comment. We will not be checking if the user owns the ENS in the schema, because we will instead verify it on the backend through the tx_receipt.